core/src/util/registry.rs

```diff
@@ -0,0 +1,114 @@
+#[cfg(target_os = "windows")]
+pub mod windows {
+    use crate::util::dsn::windows::DSNOpts;
+    use file_dbg_macros::*;
+    use std::io;
+    use winreg::{enums::HKEY_LOCAL_MACHINE, RegKey};
+
+    const ODBC: &str = "SOFTWARE\\ODBC\\ODBC.INI";
+    const ODBC_INST: &str = "SOFTWARE\\ODBC\\ODBCINST.INI";
+    const ODBC_DATA_SOURCES: &str = "SOFTWARE\\ODBC\\ODBC.INI\\ODBC Data Sources";
+
+    pub fn add_dsn(dsn_opts: &DSNOpts) -> io::Result<()> {
```

Comment:
I think that add_dsn and add_to_datasource go hand in hand. Same goes for remove_dsn and remove_from_datasource. One should be calling the other or the code should be in only one place, otherwise you might end up with a partial DSN config, no?

Reply:
Ah, I segmented them because, in my mind, the datasource is under `ODBC.INI\ODBC Data Sources`, whereas dsn is under `ODBC.INI`. This is largely gone now though since I've discovered what the installer dll is actually called (odbccp32.dll) and am using those functions.